resources/metadata/helmExecute.yaml

```diff
@@ -0,0 +1,371 @@
+        scope:
+          - GENERAL
+          - PARAMETERS
+          - STAGES
+          - STEPS
+      - name: containerImageTag
+        aliases:
+          - name: artifactVersion
+        type: string
+        description: Tag of the container which will be built - will be used together with `containerImageName` instead of parameter `containerImage`
+        scope:
+          - GENERAL
+          - PARAMETERS
+          - STAGES
+          - STEPS
+        resourceRef:
+          - name: commonPipelineEnvironment
+            param: artifactVersion
+      - name: containerRegistryUrl
+        aliases:
+          - name: dockerRegistryUrl
+        type: string
+        description: http(s) url of the Container registry where the image to deploy is located.
+        resourceRef:
+          - name: commonPipelineEnvironment
+            param: container/registryUrl
+        scope:
+          - GENERAL
+          - PARAMETERS
+          - STAGES
+          - STEPS
+        mandatory: true
+      - name: containerRegistryUser
+        description: Username for container registry access - typically provided by the CI/CD environment.
+        type: string
+        scope:
+          - PARAMETERS
+          - STAGES
+          - STEPS
+        secret: true
+        resourceRef:
+          - name: dockerCredentialsId
+            type: secret
+            param: username
+          - name: commonPipelineEnvironment
+            param: custom/repositoryUsername
+      - name: containerRegistrySecret
+        description: Name of the container registry secret used for pulling containers from the registry.
+        longDescription: |-
+          Name of the container registry secret used for pulling containers from the registry.
+
+          **For `deployTool: helm/helm3`:**<br />
+          If `containerRegistryUser` and `containerRegistryPassword` are provided, a secret is created on the fly and the information is passed to the helm template.<br />
+
+          If neither `containerRegistryUser` nor `containerRegistryPassword` are provided, it is expected that a secret with the configured name exists in the target Kubernetes cluster.<br />
+
+          **For `deployTool: kubectl`:**<br />
+          ???????If `createDockerRegistrySecret: true` and `containerRegistryUser` and `containerRegistryPassword` are provided, a secret with the given name will be created in the Kubernetes cluster unless a secret with the name already exists.
+        type: string
+        scope:
+          - PARAMETERS
+          - STAGES
+          - STEPS
+        default: regsecret
+      # - name: createDockerRegistrySecret
+      #   type: bool
+      #   description: "Only for `deployTool:kubectl`: Toggle to turn on `containerRegistrySecret` creation."
+      #   scope:
+      #     - PARAMETERS
+      #     - STAGES
+      #     - STEPS
+      #   default: false
+      - name: deploymentName
```

Comment:
not sure if we need this as mandatory parameter for all helm exeuctions.

What about e.g. `helm test`? Is it required there?